core/src/dsnp/pseudo_relationship_identifier.rs

```diff
@@ -0,0 +1,79 @@
+use crate::dsnp::dsnp_types::DsnpPrid;
+use anyhow::{Error, Result};
+use crypto_box::{aead::AeadInPlace, Nonce, PublicKey, SecretKey};
+use hkdf::Hkdf;
+use sha2::Sha256;
+use std::ops::Deref;
+use x25519_dalek::StaticSecret;
+use xsalsa20poly1305::{KeyInit, XSalsa20Poly1305};
+use zeroize::Zeroizing;
+
+const PRI_CONTEXT: &[u8] = b"PRIdCtx0";
+struct PseudonymousRelationshipIdentifier;
+
+impl PseudonymousRelationshipIdentifier {
+	fn create(
+		a: u64,
+		b: u64,
+		a_secret_key: &SecretKey,
+		b_public_key: &PublicKey,
+	) -> Result<DsnpPrid> {
+		let id_b = b.to_le_bytes();
+
+		// calculate shared secret
+		let root_shared = StaticSecret::from(*a_secret_key.as_bytes())
```

Comment:
That extra step that was happening in libsodium was to create a uniform key for encryption but since we are not using that shared key directly and deriving another key via a hash based function. This would make the final key used for encryption a uniform one.